src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -148,6 +161,40 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized String getJvmId()
+            throws AttributeNotFoundException, InstanceNotFoundException, MBeanException,
+                    ReflectionException, IOException {
+        if (!isConnected()) {
+            connect();
+        }
+        List<String> attrNames =
+                new ArrayList<>(
+                        Arrays.asList(
+                                "ClassPath",
+                                "Name",
+                                "InputArguments",
+                                "LibraryPath",
+                                "VmVendor",
+                                "VmVersion",
+                                "StartTime"));
+
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
+                DataOutputStream dos = new DataOutputStream(baos)) {
+            for (String attr : attrNames) {
+                Object attrObject =
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, ConnectionToolkit.RUNTIME_BEAN_NAME, attr));
+                if (attrObject.getClass().isArray()) {
+                    dos.writeUTF(Arrays.toString((String[]) attrObject));
```

Comment:
Ah, wait, the `(String[])` cast here might not be safe now. The `isArray()` check would pass for ex. `int[]`, but then this cast will fail. Hmm. You can use `attrObject.getClass().getComponentType()` to get the type of array element, though then you still need some way to cast it correctly.

Reply:
Compiler allows me to cast like `(Object[])` actually, I think that would be sufficient right?

Reply:
Try it with `jshell` perhaps:

```java
$ jshell
|  Welcome to JShell -- Version 17.0.4
|  For an introduction type: /help intro

jshell> var a = new int[] { 1, 2, 3}
a ==> int[3] { 1, 2, 3 }

jshell> var b = (Object[]) a
|  Error:
|  incompatible types: int[] cannot be converted to java.lang.Object[]
|  var b = (Object[]) a;
|                     ^

```

`int` is not a subclass of `Object`, so this failure makes sense.

Reply:
Ah, I didn't realize that `int[]` or all arrays are instances of Object in Java, but indeed primitives are not.